test/manipulation.test.js

```diff
@@ -1096,6 +1096,157 @@ describe('manipulation', function() {
       });
     });
   }
+  if (!getSchema().connector.replaceById) {
+    describe.skip('replaceById - not implemented', function() {});
+  } else {
+    describe('replaceOrCreate when forceId is true', function() {
+      var Post;
+      var ds = getSchema();
+      before(function(done) {
+        Post = ds.define('Post', {
+          title: {type: String, length: 255, index: true},
+          content: {type: String},
+          comments: [String],
+        }, {forceId: true});
+        ds.automigrate('Post', done);
+      });
+
+      it('fails without options on create', function(done) {
+        var post = {id: 123, title: 'a', content: 'AAA'};
+        Post.replaceOrCreate(post, function(err, p) {
+          should.exist(err);
+          err.should.match(/'Could not replace. Object with id ' + post.id + ' does not exist!'/);
+          done();
+        });
+      });
+
+      it('fails with options on create', function(done) {
+        var post = {id: 123, title: 'a', content: 'AAA'};
+        Post.replaceOrCreate(post, {validate: false}, function(err, p) {
+          should.exist(err);
+          err.should.match(/'Could not replace. Object with id ' + post.id + ' does not exist!'/);
+          done();
+        });
+      });
+
+      it('works without options on update (promise variant)', function(done) {
+        var post = {title: 'a', content: 'AAA', comments: ['Comment1']};
+        Post.create(post)
+          .then(function(created) {
+            created = created.toObject();
+            delete created.comments;
+            delete created.content;
+            created.title = 'b';
```

Comment:
Since the intention of this test is to verify how `forceId` is handled, I believe there is no need to duplicate testing of how modified/deleted properties are handled. So modify only a single property and assume other cases are already covered by existing tests.